packages/containers-shared/src/utils.ts

```diff
@@ -40,3 +41,48 @@ export const verifyDockerInstalled = async (dockerPath: string) => {
 		);
 	}
 };
+
+export function isDir(path: string) {
+	const stats = statSync(path);
+	return stats.isDirectory();
+}
+
+export const isDockerfile = (image: string): boolean => {
+	// TODO: move this into config validation
+	if (existsSync(image)) {
+		if (isDir(image)) {
+			throw new Error(
```

Comment:
all these were `UserError`s, but I replaced them with `Error` instead, since I believe it makes more sense to not ties the two together, especially in a utility function. Nothing has changed in the code that calls this fn however, so no `try...catch`. If we need to throw a `UserError` specifically, I'll adapt the code in wrangler, but wanted to get thoughts first 🙏